fast64_internal/sm64/sm64_utility.py

```diff
@@ -122,3 +136,305 @@ def convert_addr_to_func(addr: str):
+        commentless_pos += commentless.write(text[last_pos : match.start()])  # add text before comment
+        match_string = match.group(0)
+        if match_string.startswith("/"):  # actual comment
+            comment_map.append(CommentMatch(commentless_pos, len(match_string) - 1))
+            commentless_pos += commentless.write(" ")
+        else:  # stuff like strings
+            commentless_pos += commentless.write(match_string)
+        last_pos = match.end()
+
+    commentless.write(text[last_pos:])  # add any remaining text after the last match
+    return commentless.getvalue(), comment_map
+
+
+def find_descriptors(
+    text: str,
+    descriptors: list[ModifyFoundDescriptor],
+    error_if_no_header=False,
+    header: Optional[ModifyFoundDescriptor] = None,
+    error_if_no_footer=False,
+    footer: Optional[ModifyFoundDescriptor] = None,
+    ignore_comments=True,
+):
+    """Returns: The found matches mapped to the descriptors, the footer pos
+    (the end of the text if none)"""
+    if ignore_comments:
+        commentless, comment_map = get_comment_map(text)
+    else:
+        commentless, comment_map = text, []
+
+    header_matches = (
+        find_descriptor_in_text(header, commentless, comment_map, adjust=False) if header is not None else []
+    )
+    footer_matches = (
+        find_descriptor_in_text(footer, commentless, comment_map, adjust=False) if footer is not None else []
+    )
+
+    header_pos = 0
+    if len(header_matches) > 0:
+        _, header_pos, _ = header_matches[0]
+    elif header is not None and error_if_no_header:
+        raise PluginError(f"Header {header.string} does not exist.")
+
+    # find first footer after the header
+    if footer_matches:
+        if header_matches:
+            footer_pos = next((pos for _, pos, _ in footer_matches if pos >= header_pos), footer_matches[-1].start)
+        else:
+            _, footer_pos, _ = footer_matches[-1]
+    else:
+        if footer is not None and error_if_no_footer:
+            raise PluginError(f"Footer {footer.string} does not exist.")
+        footer_pos = len(commentless)
+
+    found_matches: dict[ModifyFoundDescriptor, list[DescriptorMatch]] = {}
+    for descriptor in descriptors:
+        matches = find_descriptor_in_text(descriptor, commentless, comment_map, header_pos, footer_pos)
+        if matches:
+            found_matches.setdefault(descriptor, []).extend(matches)
+    return found_matches, adjust_start_end(footer_pos, footer_pos, comment_map)[0]
+
+
+def write_or_delete_if_found(
+    path: Path,
+    to_add: Optional[list[ModifyFoundDescriptor]] = None,
+    to_remove: Optional[list[ModifyFoundDescriptor]] = None,
+    path_must_exist=False,
+    create_new=False,
+    error_if_no_header=False,
+    header: Optional[ModifyFoundDescriptor] = None,
+    error_if_no_footer=False,
+    footer: Optional[ModifyFoundDescriptor] = None,
+    ignore_comments=True,
+):
+    """
```

Comment:
You should write what the header and footer are meant to be since that is sort of the important part of this